File: app/src/main/java/androidx/viewpager/widget/ViewPagerUtils.java
package androidx.viewpager.widget;

import android.view.View;

import androidx.viewpager.widget.ViewPager;

/**
 * @author Vitaliy Markus
 */
public class ViewPagerUtils {

    public static View getCurrentView(ViewPager viewPager) {
        final int currentItem = viewPager.getCurrentItem();
        for (int i = 0; i < viewPager.getChildCount(); i++) {
            final View child = viewPager.getChildAt(i);
            final ViewPager.LayoutParams layoutParams = (ViewPager.LayoutParams) child.getLayoutParams();
            if (!layoutParams.isDecor && currentItem == layoutParams.position) {
                return child;
            }
        }
        return null;
    }
}

File: app/src/main/java/androidx/viewpager/widget/ViewPagerUtils.kt
@file:JvmName("ViewPagerUtils")

package androidx.viewpager.widget

import android.view.View

/**
 * @author Vitaliy Markus
 */

fun ViewPager.getCurrentView(): View? {
    val currentItem = currentItem
    for (i in 0 until childCount) {
        val child = getChildAt(i)
        val layoutParams = child.layoutParams as ViewPager.LayoutParams
        if (!layoutParams.isDecor && currentItem == layoutParams.position) {
            return child
        }
    }
    return null
}

fun ViewPager.getView(position: Int): View? {
    for (i in 0 until childCount) {
        val child = getChildAt(i)
        val layoutParams = child.layoutParams as ViewPager.LayoutParams
        if (!layoutParams.isDecor && position == layoutParams.position) {
            return child
        }
    }
    return null
}


